core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -446,6 +446,16 @@ public QuerySearchResult executeQueryPhase(QuerySearchRequest request) {
         }
     }
 
+    private boolean fetchPhaseShouldFreeContext(SearchContext context) {
+        if (context.scrollContext() == null) {
+            // simple search, no scroll
+            return true;
+        } else {
+            // scroll request, but the scroll was not extended
```

Comment:
oops, nevermind I misunderstood something...

I do wonder if maybe should destroy the SearchContext if all hits of a shard have been consumed in a followup issue?